community/modules/compute/gke-node-pool/main.tf

```diff
@@ -0,0 +1,141 @@
+/**
+  * Copyright 2023 Google LLC
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+locals {
+  sa_email = var.service_account.email != null ? var.service_account.email : data.google_compute_default_service_account.default_sa.email
+}
+
+data "google_compute_default_service_account" "default_sa" {
+  project = var.project_id
+}
+
+resource "google_container_node_pool" "node_pool" {
+  provider = google-beta
+
+  name    = var.name == null ? var.machine_type : var.name
+  cluster = var.cluster_id
+  autoscaling {
+    total_min_node_count = var.total_min_nodes
+    total_max_node_count = var.total_max_nodes
+    location_policy      = "ANY"
+  }
+
+  upgrade_settings {
+    strategy        = "SURGE"
+    max_surge       = 0
+    max_unavailable = 20
+  }
+
+  management {
+    auto_repair  = true
+    auto_upgrade = true
```

Comment:
shouldn't this be disabled by default?

Reply:
From description:  
- Resolve `auto_upgrade`: conflicting recommendations to have `auto_upgrade=false` and `release_channel=regular`

If release_channel is regular then you have to enable auto upgrade. The recommendation was to use the regular release channel.  I have added it to the list of things to rectify with GKE team. In general I will need to take another pass at the version management strategy.